src/sonic_ax_impl/mibs/ieee802_1ab.py

```diff
@@ -511,28 +510,26 @@ def update_rem_if_mgmt(self, if_oid, if_name):
             if len(mgmt_ip_str) == 0:
                 # the peer advertise an emtpy mgmt address
                 return
-            time_mark = int(lldp_kvs['lldp_rem_time_mark'])
-            remote_index = int(lldp_kvs['lldp_rem_index'])
-            subtype = self.get_subtype(mgmt_ip_str)
-            ip_hex = self.get_ip_hex(mgmt_ip_str, subtype)
-            if subtype == ManAddrConst.man_addr_subtype_ipv4:
-                addr_subtype_sub_oid = 4
-                mgmt_ip_sub_oid = (addr_subtype_sub_oid, *[int(i) for i in mgmt_ip_str.split('.')])
-            elif subtype == ManAddrConst.man_addr_subtype_ipv6:
-                addr_subtype_sub_oid = 6
-                mgmt_ip_sub_oid = (addr_subtype_sub_oid, *[int(i, 16) if i else 0 for i in mgmt_ip_str.split(':')])
-            else:
-                logger.warning("Invalid management IP {}".format(mgmt_ip_str))
-                return
-            self.if_range.append((time_mark,
-                                  if_oid,
-                                  remote_index,
-                                  subtype,
-                                  *mgmt_ip_sub_oid))
-
-            self.mgmt_ips.update({if_name: {"ip_str": mgmt_ip_str,
-                                            "addr_subtype": subtype,
-                                            "addr_hex": ip_hex}})
+            for mgmt_ip in set(mgmt_ip_str.split(',')):
```

Comment:
>set [](start = 27, length = 3)

What is the benefit of `set` ? Just use splited list?

Reply:
this is just extra check in case if we have same ip's

Reply:
Please be aware that different ipv6 strings could be the same ipv6 address.

---
In reply to: [593529516](https://github.com/Azure/sonic-snmpagent/pull/201#discussion_r593529516) [](ancestors = 593529516)

Reply:
@qiluo-msft Updated.